sinks/datadog/datadog.go

```diff
@@ -108,7 +108,21 @@ func (dd *DatadogMetricSink) Flush(ctx context.Context, interMetrics []samplers.
 	span, _ := trace.StartSpanFromContext(ctx, "")
 	defer span.ClientFinish(dd.traceClient)
 
-	ddmetrics := dd.finalizeMetrics(interMetrics)
+	ddmetrics, checks := dd.finalizeMetrics(interMetrics)
+
+	if len(checks) != 0 {
+		// this endpoint is not documented to take an array... but it does
+		// another curious constraint of this endpoint is that it does not
+		// support "Content-Encoding: deflate"
+		err := vhttp.PostHelper(context.TODO(), dd.HTTPClient, dd.traceClient, http.MethodPost, fmt.Sprintf("%s/api/v1/check_run?api_key=%s", dd.DDHostname, dd.APIKey), checks, "flush_checks", false, map[string]string{"sink": "datadog"}, dd.log)
```

Comment:
you want `context.Background()` here!

Reply:
oh wow... should have scanned everything for a TODO. I guess I've just been in the mode of least risk possible but that pretty obviously doesn't belong here.